YouTubeSearch/src/redux/reducers/homeVideoReducer.ts

```diff
@@ -0,0 +1,42 @@
+import {
+  HOME_VIDEOS_FAIL,
+  HOME_VIDEOS_REQUEST,
+  HOME_VIDEOS_SUCCESS,
+} from '../actionType';
+
+const homeVideosReducer = (

```

Comment:
в каждом `reducer` определи тип для `state`,чтобы избегать использование `any`
например для `homeVideosReducer`
```suggestion

interface HomeVideoState {
  videos: VideoItem[];
  loading: boolean;
  nextPageToken?: string;
}

const initialState: HomeVideoState = {
  videos: [],
  loading: false,
  nextPageToken: undefined,
}

const homeVideosReducer = (
  state: HomeVideoState = initialState,
```
Все интерфейсы для объектов, которые ты получаешь по API определи в `./src/Api/interfaces.ts`
например 
```
interface VideoItem { 
  id: string;
  snippet: VideoItemSnippet;
}

interface VideoItemSnippet {
  title: string;
  publishedAt: string;
  thumbnails: {
    medium: {
      url: string;
    }
  },
}
```
Если у тебя поле может быть `undefined` , то ты можешь сделать его `optional` через `?` -`someField?: string;`

Если у тебя значение может быть разных типов, то перечисли их через `|` - `someField: string | number;`

Reply:
используй пожалуйста `handleActions` из `redux-actions`.
https://thanarie.gitbooks.io/typescript-webpack-book/content/redux-actions.html

```suggestion

type CombinedPayloads = FetchHomeVideoPayload | ...

const homeVideosReducer = handleActions<HomeVideoState, CombinedPayloads>({
  [FETCH_HOME_VIDEOS_REQUEST]: (state, { payload }: Action<FetchHomeVideoPayload>): HomeVideoState => ({
    ...state,
    videos: [...state.videos, ...payload.videos],
    loading: false,
    nextPageToken: payload.nextPageToken,
  }),
  ...
}, initialState);

```